### ISSUE TITLE
Async hooks continue running after stopping a service

### ISSUE THREAD

**Issue by davidMcneil** [habitat-sh/habitat#7128]
Async hooks continue running even after stopping a service because they are executed in a new thread. A fix would involve creating a new [`cancelable_future`](https://github.com/habitat-sh/habitat/blob/4c4ae04137c632563abcb247c60db9600bad0597/components/sup/src/sup_futures.rs#L26) that tracks the PID and kills the process when `terminate` is called.

The [`tokio_process`](https://docs.rs/tokio-process/0.3.0-alpha.2/tokio_process/) crate is specifically designed to handle this, but would require changes to the way windows process are handled per this [comment](https://github.com/habitat-sh/habitat/blob/4c4ae04137c632563abcb247c60db9600bad0597/components/sup/src/manager/service/hook_runner.rs#L4).

**Comment by mwrock** [comment]
Yeah the big dilemma we face with Rust's `process` API and thus `tokio_process` as well is that it does not support the Windows APIs that allow you to spawn a process as an alternate user. Its limited to starting the process as the current user.
